<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Workshop of microbiome dataset analysis using MicrobiotaProcess â€¢ MicrobiotaProcessWorkshop</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Workshop of microbiome dataset analysis using MicrobiotaProcess">
<meta property="og:description" content="MicrobiotaProcessWorkshop">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MicrobiotaProcessWorkshop</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MicrobiotaProcessWorkshop.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/xiangpin/MicrobiotaProcessWorkshop">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="MicrobiotaProcessWorkshop_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Workshop of microbiome dataset analysis using MicrobiotaProcess</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/xiangpin/MicrobiotaProcessWorkshop/blob/master/vignettes/MicrobiotaProcessWorkshop.Rmd"><code>vignettes/MicrobiotaProcessWorkshop.Rmd</code></a></small>
      <div class="hidden name"><code>MicrobiotaProcessWorkshop.Rmd</code></div>

    </div>

    
    
<div id="load-required-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#load-required-packages" class="anchor"></a>Load required packages</h2>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span>
    <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/YuLab-SMU/MicrobiotaProcess/">MicrobiotaProcess</a></span><span class="op">)</span>
    <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://dx.plos.org/10.1371/journal.pone.0061217">phyloseq</a></span><span class="op">)</span>
    <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
    <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span></pre></div>
<p>Here, we use the 43 pediatric IBD stool samples as example, obtained from the Integrative Human Microbiome Project Consortium (iHMP).</p>
</div>
<div id="importing-the-output-of-dada2" class="section level2">
<h2 class="hasAnchor">
<a href="#importing-the-output-of-dada2" class="anchor"></a>Importing the output of dada2</h2>
<p>The datasets were obtained from <code>https://www.microbiomeanalyst.ca/MicrobiomeAnalyst/resources/data/ibd_data.zip</code>. It contains <code>ibd_asv_table.txt</code>, which is feature table (<em>row features</em> X <em>column samples</em>), <code>ibd_meta.csv</code>, which is metadata file of samples, and <code>ibd_taxa.txt</code>, which is the taxonomic annotation of features. In the session, we use <code>import_dada2</code> of <code>MicrobiotaProcess</code> to import the datasets, and return a <code>phyloseq</code> object.</p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/YuLab-SMU/MicrobiotaProcess/">MicrobiotaProcess</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://dx.plos.org/10.1371/journal.pone.0061217">phyloseq</a></span><span class="op">)</span>
<span class="va">otuda</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="st">"./IBD_data/ibd_asv_table.txt"</span>, header<span class="op">=</span><span class="cn">T</span>, 
                    check.names<span class="op">=</span><span class="cn">F</span>, comment.char<span class="op">=</span><span class="st">""</span>, row.names<span class="op">=</span><span class="fl">1</span>, sep<span class="op">=</span><span class="st">"\t"</span><span class="op">)</span>
<span class="co"># building the output format of removeBimeraDenovo of dada2</span>
<span class="va">otuda</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transpose-methods.html">t</a></span><span class="op">(</span><span class="va">otuda</span><span class="op">)</span>, check.names<span class="op">=</span><span class="cn">F</span><span class="op">)</span>
<span class="va">sampleda</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span><span class="st">"./IBD_data/ibd_meta.csv"</span>, row.names<span class="op">=</span><span class="fl">1</span>, comment.char<span class="op">=</span><span class="st">""</span><span class="op">)</span>
<span class="va">taxda</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="st">"./IBD_data/ibd_taxa.txt"</span>, header<span class="op">=</span><span class="cn">T</span>, 
                    row.names<span class="op">=</span><span class="fl">1</span>, check.names<span class="op">=</span><span class="cn">F</span>, comment.char<span class="op">=</span><span class="st">""</span><span class="op">)</span>
<span class="co"># the feature names should be the same with rownames of taxda. </span>
<span class="va">taxda</span> <span class="op">&lt;-</span> <span class="va">taxda</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">otuda</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">taxda</span><span class="op">)</span><span class="op">)</span>,<span class="op">]</span>
<span class="va">psraw</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MicrobiotaProcess/man/import_dada2.html">import_dada2</a></span><span class="op">(</span>seqtab<span class="op">=</span><span class="va">otuda</span>, taxatab<span class="op">=</span><span class="va">taxda</span>, sampleda<span class="op">=</span><span class="va">sampleda</span><span class="op">)</span>
<span class="co"># view the reads depth of samples. In this example, </span>
<span class="co"># we remove the sample contained less than 20914 otus.</span>
<span class="co"># sort(rowSums(otu_table(psraw)))</span>

<span class="co"># samples were removed, which the reads number is too little.</span>
<span class="va">psraw</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/prune_samples-methods.html">prune_samples</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_sums.html">sample_sums</a></span><span class="op">(</span><span class="va">psraw</span><span class="op">)</span><span class="op">&gt;=</span><span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html">otu_table</a></span><span class="op">(</span><span class="va">psraw</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span>, <span class="va">psraw</span><span class="op">)</span>
<span class="co"># then the samples were rarefied to 20914 reads.</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1024</span><span class="op">)</span>
<span class="va">ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/rarefy_even_depth.html">rarefy_even_depth</a></span><span class="op">(</span><span class="va">psraw</span><span class="op">)</span>
<span class="va">ps</span>
<span class="co">#&gt; phyloseq-class experiment-level object</span>
<span class="co">#&gt; otu_table()   OTU Table:         [ 1432 taxa and 41 samples ]</span>
<span class="co">#&gt; sample_data() Sample Data:       [ 41 samples by 1 sample variables ]</span>
<span class="co">#&gt; tax_table()   Taxonomy Table:    [ 1432 taxa by 7 taxonomic ranks ]</span>
<span class="co">#&gt; refseq()      DNAStringSet:      [ 1432 reference sequences ]</span></pre></div>
</div>
<div id="example-of-rarefaction-visualization" class="section level2">
<h2 class="hasAnchor">
<a href="#example-of-rarefaction-visualization" class="anchor"></a>Example of rarefaction visualization</h2>
<p>Rarefaction, based on sampling technique, was used to compensate for the effect of sample size on the number of units observed in a sample. <code>MicrobiotaProcess</code> provided <code>ggrarecurve</code> to plot the curves, based on <code>rrarefy</code> of <code>vegan</code></p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/YuLab-SMU/MicrobiotaProcess/">MicrobiotaProcess</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com">patchwork</a></span><span class="op">)</span>
<span class="co"># for reproducibly random number</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1024</span><span class="op">)</span>
<span class="va">p_rare</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MicrobiotaProcess/man/ggrarecurve.html">ggrarecurve</a></span><span class="op">(</span>obj<span class="op">=</span><span class="va">ps</span>,
                      indexNames<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Observe"</span>,<span class="st">"Chao1"</span>,<span class="st">"ACE"</span><span class="op">)</span>,
                      chunks<span class="op">=</span><span class="fl">400</span><span class="op">)</span> <span class="op">+</span>
          <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.spacing.y<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">0.01</span>,<span class="st">"cm"</span><span class="op">)</span>,
                legend.text<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1024</span><span class="op">)</span>
<span class="va">prare1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MicrobiotaProcess/man/ggrarecurve.html">ggrarecurve</a></span><span class="op">(</span>obj<span class="op">=</span><span class="va">ps</span>, factorNames<span class="op">=</span><span class="st">"Group"</span>,
                      indexNames<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Observe"</span>, <span class="st">"Chao1"</span>, <span class="st">"ACE"</span><span class="op">)</span>,
                      chunks<span class="op">=</span><span class="fl">400</span>, se<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
          <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#00AED7"</span>, <span class="st">"#FD9347"</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
          <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#00AED7"</span>, <span class="st">"#FD9347"</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
          <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span>
          <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.text<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">8</span><span class="op">)</span>, panel.grid<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,
                strip.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_rect</a></span><span class="op">(</span>colour<span class="op">=</span><span class="cn">NA</span>,fill<span class="op">=</span><span class="st">"grey"</span><span class="op">)</span>,
                strip.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span><span class="op">)</span>
<span class="va">p_rare</span> <span class="op">/</span> <span class="va">prare1</span></pre></div>
<div class="figure">
<img src="MicrobiotaProcessWorkshop_files/figure-html/rarecurve-1.png" alt="\label{fig:Rarecurve}Example of rarefaction curve of samples" width="912"><p class="caption">
Example of rarefaction curve of samples
</p>
</div>
<p>Since the curves in each sample were near saturation, the sequencing data were great enough with very few new species undetected</p>
</div>
<div id="example-of-calculate-alpha-index-and-visualization" class="section level2">
<h2 class="hasAnchor">
<a href="#example-of-calculate-alpha-index-and-visualization" class="anchor"></a>Example of calculate alpha index and visualization</h2>
<p>Alpha index can evaluate the richness and abundance of microbial communities. <code>MicrobiotaProcess</code> provides <code>get_alphaindex</code> to calculate alpha index. Six common diversity measures (<em>Observe</em>, <em>Chao1</em>, <em>ACE</em>, <em>Shannon</em>, <em>Simpson</em>, <em>J</em>) are supported. And the different groups of samples can be tested and visualize by <code>ggbox</code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/YuLab-SMU/MicrobiotaProcess/">MicrobiotaProcess</a></span><span class="op">)</span>
<span class="va">alphaobj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MicrobiotaProcess/man/get_alphaindex.html">get_alphaindex</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">alphaobj</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt;         Observe    Chao1       ACE  Shannon   Simpson         J   Group</span>
<span class="co">#&gt; S206700      86  93.0000  90.13504 2.868408 0.8911390 0.6439570      CD</span>
<span class="co">#&gt; S206701      45  45.0000  45.00000 1.634528 0.5614782 0.4293861      CD</span>
<span class="co">#&gt; S206702      38  39.5000  39.31896 0.509799 0.1502078 0.1401476      CD</span>
<span class="co">#&gt; S206703     156 157.6667 157.89265 3.507910 0.9104597 0.6946554 Control</span>
<span class="co">#&gt; S206704     155 160.0000 157.08120 3.946515 0.9642167 0.7825068 Control</span>
<span class="co">#&gt; S206708     113 118.0000 114.51622 2.747711 0.8443571 0.5812324      CD</span>
<span class="va">p_alpha</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MicrobiotaProcess/man/ggbox.html">ggbox</a></span><span class="op">(</span><span class="va">alphaobj</span>, geom<span class="op">=</span><span class="st">"violin"</span>, factorNames<span class="op">=</span><span class="st">"Group"</span><span class="op">)</span> <span class="op">+</span>
           <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#00AED7"</span>, <span class="st">"#FD9347"</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
       <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>strip.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_rect</a></span><span class="op">(</span>colour<span class="op">=</span><span class="cn">NA</span>, fill<span class="op">=</span><span class="st">"grey"</span><span class="op">)</span><span class="op">)</span>
<span class="va">p_alpha</span></pre></div>
<div class="figure">
<img src="MicrobiotaProcessWorkshop_files/figure-html/alphatest-1.png" alt="\label{fig:Alphabox}Example of alpha box of samples" width="768"><p class="caption">
Example of alpha box of samples
</p>
</div>
</div>
<div id="example-of-visualization-of-taxonomy-abundance" class="section level2">
<h2 class="hasAnchor">
<a href="#example-of-visualization-of-taxonomy-abundance" class="anchor"></a>Example of visualization of taxonomy abundance</h2>
<p><code>MicrobiotaProcess</code> presents the <code>ggbartax</code> for the visualization of composition of microbial communities. If you want to get the abundance of specific levels of class, You can use <code>get_taxadf</code> then use <code>ggbartax</code> to visualize.</p>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/YuLab-SMU/MicrobiotaProcess/">MicrobiotaProcess</a></span><span class="op">)</span>

<span class="va">classtaxa</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MicrobiotaProcess/man/get_taxadf.html">get_taxadf</a></span><span class="op">(</span>obj<span class="op">=</span><span class="va">ps</span>, taxlevel<span class="op">=</span><span class="fl">3</span><span class="op">)</span>
<span class="co"># The 30 most abundant taxonomy will be visualized by default (parameter `topn=30`). </span>
<span class="va">pclass</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MicrobiotaProcess/man/ggbartax.html">ggbartax</a></span><span class="op">(</span>obj<span class="op">=</span><span class="va">classtaxa</span>, facetNames<span class="op">=</span><span class="st">"Group"</span><span class="op">)</span> <span class="op">+</span>
          <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span>
          <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"relative abundance (%)"</span><span class="op">)</span> <span class="op">+</span>
          <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html">colorRampPalette</a></span><span class="op">(</span><span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span><span class="op">(</span><span class="fl">12</span>,<span class="st">"Set3"</span><span class="op">)</span><span class="op">)</span><span class="op">(</span><span class="fl">31</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
          <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html">guides</a></span><span class="op">(</span>fill<span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html">guide_legend</a></span><span class="op">(</span>keywidth <span class="op">=</span> <span class="fl">0.5</span>, keyheight <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span>
<span class="va">pclass</span></pre></div>
<div class="figure">
<img src="MicrobiotaProcessWorkshop_files/figure-html/abundance-1.png" alt="\label{fig:Abundance1}Example of taxonomic abundance of samples" width="816"><p class="caption">
Example of taxonomic abundance of samples
</p>
</div>
<p>If you want to view the count of taxonomy, you can set the <code>count</code> parameter of <code>ggbartax</code> to <code>TRUE</code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="va">pclass2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MicrobiotaProcess/man/ggbartax.html">ggbartax</a></span><span class="op">(</span>obj<span class="op">=</span><span class="va">classtaxa</span>, count<span class="op">=</span><span class="cn">TRUE</span>, facetNames<span class="op">=</span><span class="st">"Group"</span><span class="op">)</span> <span class="op">+</span>
          <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span>
          <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"count reads"</span><span class="op">)</span> <span class="op">+</span>
          <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html">colorRampPalette</a></span><span class="op">(</span><span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span><span class="op">(</span><span class="fl">12</span>,<span class="st">"Set3"</span><span class="op">)</span><span class="op">)</span><span class="op">(</span><span class="fl">31</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
          <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html">guides</a></span><span class="op">(</span>fill<span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html">guide_legend</a></span><span class="op">(</span>keywidth <span class="op">=</span> <span class="fl">0.5</span>, keyheight <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span>
<span class="va">pclass2</span></pre></div>
<div class="figure">
<img src="MicrobiotaProcessWorkshop_files/figure-html/abundance2-1.png" alt="\label{fig:Abundance2}Example of taxonomic abundance (Count) of samples" width="816"><p class="caption">
Example of taxonomic abundance (Count) of samples
</p>
</div>
</div>
<div id="example-of-pca-and-pcoa-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#example-of-pca-and-pcoa-analysis" class="anchor"></a>Example of PCA and PCoA analysis</h2>
<p><code>PCA</code> (Principal component analysis) and <code>PCoA</code> (Principal Coordinate Analysis) are general statistical procedures to compare groups of samples. And <code>PCoA</code> can based on the phylogenetic or count-based distance metrics, such as <code>Bray-Curtis</code>, <code>Jaccard</code>, <code>Unweighted-UniFrac</code> and <code>weighted-UniFrac</code>. <code>MicrobiotaProcess</code> presents the <code>get_pca</code>, <code>get_pcoa</code> and <code>ggordpoint</code> for the analysis.</p>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/YuLab-SMU/MicrobiotaProcess/">MicrobiotaProcess</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com">patchwork</a></span><span class="op">)</span>
<span class="co"># If the input was normalized, the method parameter should be setted NULL.</span>
<span class="va">pcares</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MicrobiotaProcess/man/get_pca.html">get_pca</a></span><span class="op">(</span>obj<span class="op">=</span><span class="va">ps</span>, method<span class="op">=</span><span class="st">"hellinger"</span><span class="op">)</span>
<span class="co"># Visulizing the result</span>
<span class="va">pcaplot1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MicrobiotaProcess/man/ggordpoint.html">ggordpoint</a></span><span class="op">(</span>obj<span class="op">=</span><span class="va">pcares</span>, biplot<span class="op">=</span><span class="cn">TRUE</span>, speciesannot<span class="op">=</span><span class="cn">TRUE</span>,
                      factorNames<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Group"</span><span class="op">)</span>, ellipse<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span>
            <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#00AED7"</span>, <span class="st">"#FD9347"</span><span class="op">)</span><span class="op">)</span>
<span class="co"># pc = c(1, 3) to show the first and third principal components.</span>
<span class="va">pcaplot2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MicrobiotaProcess/man/ggordpoint.html">ggordpoint</a></span><span class="op">(</span>obj<span class="op">=</span><span class="va">pcares</span>, pc<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">3</span><span class="op">)</span>, biplot<span class="op">=</span><span class="cn">TRUE</span>, speciesannot<span class="op">=</span><span class="cn">TRUE</span>,
                       factorNames<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Group"</span><span class="op">)</span>, ellipse<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span>
            <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#00AED7"</span>, <span class="st">"#FD9347"</span><span class="op">)</span><span class="op">)</span>
<span class="va">pcaplot1</span> <span class="op">|</span> <span class="va">pcaplot2</span></pre></div>
<div class="figure">
<img src="MicrobiotaProcessWorkshop_files/figure-html/PCAplot-1.png" alt="\label{fig:PCA}Example of PCA of samples" width="864"><p class="caption">
Example of PCA of samples
</p>
</div>
</div>
<div id="example-of-pcoa-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#example-of-pcoa-analysis" class="anchor"></a>Example of PCoA analysis</h2>
<div class="sourceCode" id="cb8"><pre class="downlit">
<span class="va">pcoares</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MicrobiotaProcess/man/get_pcoa.html">get_pcoa</a></span><span class="op">(</span>obj<span class="op">=</span><span class="va">ps</span>, distmethod<span class="op">=</span><span class="st">"bray"</span>, method<span class="op">=</span><span class="st">"hellinger"</span><span class="op">)</span>
<span class="co"># Visualizing the result</span>
<span class="va">pcoaplot1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MicrobiotaProcess/man/ggordpoint.html">ggordpoint</a></span><span class="op">(</span>obj<span class="op">=</span><span class="va">pcoares</span>, biplot<span class="op">=</span><span class="cn">TRUE</span>, speciesannot<span class="op">=</span><span class="cn">TRUE</span>,
                       factorNames<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Group"</span><span class="op">)</span>, ellipse<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span>
            <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#00AED7"</span>, <span class="st">"#FD9347"</span><span class="op">)</span><span class="op">)</span>
<span class="co"># first and third principal co-ordinates</span>
<span class="va">pcoaplot2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MicrobiotaProcess/man/ggordpoint.html">ggordpoint</a></span><span class="op">(</span>obj<span class="op">=</span><span class="va">pcoares</span>, pc<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">3</span><span class="op">)</span>, biplot<span class="op">=</span><span class="cn">TRUE</span>, speciesannot<span class="op">=</span><span class="cn">TRUE</span>,
                        factorNames<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Group"</span><span class="op">)</span>, ellipse<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span>
             <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#00AED7"</span>, <span class="st">"#FD9347"</span><span class="op">)</span><span class="op">)</span>
<span class="va">pcoaplot1</span> <span class="op">|</span> <span class="va">pcoaplot2</span></pre></div>
<div class="figure">
<img src="MicrobiotaProcessWorkshop_files/figure-html/PCoAplot-1.png" alt="\label{fig:PCoA}Example of PCoA of samples" width="864"><p class="caption">
Example of PCoA of samples
</p>
</div>
</div>
<div id="example-of-hierarchical-cluster-analysis-of-samples" class="section level2">
<h2 class="hasAnchor">
<a href="#example-of-hierarchical-cluster-analysis-of-samples" class="anchor"></a>Example of hierarchical cluster analysis of samples</h2>
<p><code>beta diversity</code> metrics can assess the differences between microbial communities. It can be visualized with <code>PCA</code> or <code>PCoA</code>, this can also be visualized with hierarchical clustering. <code>MicrobiotaProcess</code> implements the analysis based on <code>ggtree</code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/YuLab-SMU/MicrobiotaProcess/">MicrobiotaProcess</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/treedata-book/">ggtree</a></span><span class="op">)</span>
<span class="va">hcsample</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MicrobiotaProcess/man/get_clust.html">get_clust</a></span><span class="op">(</span>obj<span class="op">=</span><span class="va">ps</span>, distmethod<span class="op">=</span><span class="st">"bray"</span>,
                      method<span class="op">=</span><span class="st">"hellinger"</span>, hclustmethod<span class="op">=</span><span class="st">"average"</span><span class="op">)</span>
<span class="co"># rectangular layout</span>
<span class="va">cplot1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MicrobiotaProcess/man/ggclust.html">ggclust</a></span><span class="op">(</span>obj<span class="op">=</span><span class="va">hcsample</span>,
                  layout <span class="op">=</span> <span class="st">"rectangular"</span>,
                  pointsize<span class="op">=</span><span class="fl">1</span>,
                  fontsize<span class="op">=</span><span class="fl">0</span>,
                  factorNames<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Group"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
              <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#00AED7"</span>, <span class="st">"#FD9347"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
          <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/theme_tree2.html">theme_tree2</a></span><span class="op">(</span>legend.position<span class="op">=</span><span class="st">"right"</span>,
              plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face<span class="op">=</span><span class="st">"bold"</span>, lineheight<span class="op">=</span><span class="fl">25</span>,hjust<span class="op">=</span><span class="fl">0.5</span><span class="op">)</span><span class="op">)</span>
<span class="co"># circular layout</span>
<span class="va">cplot2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MicrobiotaProcess/man/ggclust.html">ggclust</a></span><span class="op">(</span>obj<span class="op">=</span><span class="va">hcsample</span>,
                  layout <span class="op">=</span> <span class="st">"circular"</span>,
              pointsize<span class="op">=</span><span class="fl">1</span>,
              fontsize<span class="op">=</span><span class="fl">2</span>,
             factorNames<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Group"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
          <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#00AED7"</span>, <span class="st">"#FD9347"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
          <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position<span class="op">=</span><span class="st">"right"</span><span class="op">)</span>
<span class="va">cplot1</span> <span class="op">|</span> <span class="va">cplot2</span></pre></div>
<div class="figure">
<img src="MicrobiotaProcessWorkshop_files/figure-html/Clust-1.png" alt="\label{fig:HClust}Example of hierarchical clustering of samples" width="960"><p class="caption">
Example of hierarchical clustering of samples
</p>
</div>
</div>
<div id="example-of-biomarker-discovery" class="section level2">
<h2 class="hasAnchor">
<a href="#example-of-biomarker-discovery" class="anchor"></a>Example of biomarker discovery</h2>
<p>This package provides <code>diff_analysis</code> to detect the biomarker. And the results can be visualized by <code>ggdiffbox</code>, <code>ggdiffclade</code>, <code>ggeffectsize</code> and <code>ggdifftaxbar</code></p>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="co"># for the kruskal_test and wilcox_test</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://coin.r-forge.r-project.org">coin</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/YuLab-SMU/MicrobiotaProcess/">MicrobiotaProcess</a></span><span class="op">)</span>
<span class="co"># Since the effect size was calculated by randomly re-sampling, </span>
<span class="co"># the seed should be set for reproducibly results.</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1024</span><span class="op">)</span>
<span class="va">deres</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MicrobiotaProcess/man/diff_analysis.html">diff_analysis</a></span><span class="op">(</span>obj <span class="op">=</span> <span class="va">ps</span>, classgroup <span class="op">=</span> <span class="st">"Group"</span>,
                       mlfun <span class="op">=</span> <span class="st">"lda"</span>,
                       filtermod <span class="op">=</span> <span class="st">"pvalue"</span>,
                       firstcomfun <span class="op">=</span> <span class="st">"kruskal_test"</span>,
                       firstalpha <span class="op">=</span> <span class="fl">0.05</span>,
                       strictmod <span class="op">=</span> <span class="cn">TRUE</span>,
                       secondcomfun <span class="op">=</span> <span class="st">"wilcox_test"</span>,
                       subclmin <span class="op">=</span> <span class="fl">3</span>,
                       subclwilc <span class="op">=</span> <span class="cn">TRUE</span>,
                       secondalpha <span class="op">=</span> <span class="fl">0.01</span>,
                       lda<span class="op">=</span><span class="fl">3</span><span class="op">)</span>
<span class="va">deres</span>
<span class="co">#&gt; The original data: 689 features and 41 samples</span>
<span class="co">#&gt; The sample data: 1 variables and 41 samples</span>
<span class="co">#&gt; The taxda contained 1432 by 7 rank</span>
<span class="co">#&gt; after first test (kruskal_test) number of feature (pvalue&lt;=0.05):71</span>
<span class="co">#&gt; after second test (wilcox_test) number of significantly discriminative feature:28</span>
<span class="co">#&gt; after lda, Number of discriminative features: 22 (certain taxonomy classification:15; uncertain taxonomy classication: 7)</span></pre></div>
</div>
<div id="example-of-visualization-of-different-results-by-ggdiffclade" class="section level2">
<h2 class="hasAnchor">
<a href="#example-of-visualization-of-different-results-by-ggdiffclade" class="anchor"></a>Example of visualization of different results by <code>ggdiffclade</code>
</h2>
<p>The color of discriminative taxa represent the taxa is more abundant in the corresponding group. The point size shows the negative logarithms (base 10) of pvalue. The bigger size of point shows more significant (lower pvalue), the pvalue was calculated in the first step test.</p>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="va">diffclade_p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MicrobiotaProcess/man/ggdiffclade.html">ggdiffclade</a></span><span class="op">(</span>
                   obj<span class="op">=</span><span class="va">deres</span>, 
                   alpha<span class="op">=</span><span class="fl">0.3</span>, 
                   linewd<span class="op">=</span><span class="fl">0.15</span>,
                   skpointsize<span class="op">=</span><span class="fl">0.6</span>, 
                   layout<span class="op">=</span><span class="st">"radial"</span>,
                   taxlevel<span class="op">=</span><span class="fl">3</span>, 
                   removeUnkown<span class="op">=</span><span class="cn">TRUE</span>,
               <span class="op">)</span> <span class="op">+</span>
               <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>
                   values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#00AED7"</span>, <span class="st">"#FD9347"</span><span class="op">)</span>
               <span class="op">)</span> <span class="op">+</span>
               <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html">guides</a></span><span class="op">(</span>color <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html">guide_legend</a></span><span class="op">(</span>
                                  keywidth <span class="op">=</span> <span class="fl">0.1</span>, 
                                  keyheight <span class="op">=</span> <span class="fl">0.6</span>,
                                  order <span class="op">=</span> <span class="fl">3</span>,
                                  ncol<span class="op">=</span><span class="fl">1</span><span class="op">)</span>
               <span class="op">)</span> <span class="op">+</span>
               <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>
                   panel.background<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_rect</a></span><span class="op">(</span>fill<span class="op">=</span><span class="cn">NA</span><span class="op">)</span>,
                   legend.position<span class="op">=</span><span class="st">"right"</span>, 
                   plot.margin<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">margin</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span>,<span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>,
                   legend.spacing.y<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">0.02</span>, <span class="st">"cm"</span><span class="op">)</span>, 
                   legend.title<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">7</span><span class="op">)</span>,
                   legend.text<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">6</span><span class="op">)</span>, 
                   legend.box.spacing<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">0.02</span>,<span class="st">"cm"</span><span class="op">)</span>
               <span class="op">)</span>
<span class="va">diffclade_p</span></pre></div>
<div class="figure">
<img src="MicrobiotaProcessWorkshop_files/figure-html/plotclade-1.png" alt="\label{fig:clade}Example of different biomarker clade" width="576"><p class="caption">
Example of different biomarker clade
</p>
</div>
</div>
<div id="example-of-visualization-of-different-results-by-ggdiffbox" class="section level2">
<h2 class="hasAnchor">
<a href="#example-of-visualization-of-different-results-by-ggdiffbox" class="anchor"></a>Example of visualization of different results by <code>ggdiffbox</code>
</h2>
<p>The left panel represents the relative abundance or abundance (according the standard_method) of biomarker, the right panel represents the confident interval of effect size (LDA or MDA) of biomarker. The bigger confident interval shows that the biomarker is more fluctuant, owing to the influence of samples number.</p>
<div class="sourceCode" id="cb12"><pre class="downlit">
<span class="va">diffbox</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MicrobiotaProcess/man/ggdiffbox.html">ggdiffbox</a></span><span class="op">(</span>obj<span class="op">=</span><span class="va">deres</span>, box_notch<span class="op">=</span><span class="cn">FALSE</span>, 
             colorlist<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#00AED7"</span>, <span class="st">"#FD9347"</span><span class="op">)</span>, l_xlabtext<span class="op">=</span><span class="st">"relative abundance"</span><span class="op">)</span>
<span class="va">diffbox</span></pre></div>
<div class="figure">
<img src="MicrobiotaProcessWorkshop_files/figure-html/diffbox-1.png" alt="\label{fig:diffboxplot}Example of biomarker boxplot and effect size" width="576"><p class="caption">
Example of biomarker boxplot and effect size
</p>
</div>
</div>
<div id="example-of-visualization-of-different-results-by-ggdifftaxbar" class="section level2">
<h2 class="hasAnchor">
<a href="#example-of-visualization-of-different-results-by-ggdifftaxbar" class="anchor"></a>Example of visualization of different results by <code>ggdifftaxbar</code>
</h2>
<p><code>ggdifftaxbar</code> can visualize the abundance of biomarker in each samples of groups, the mean and median abundance of groups or subgroups are also showed. <code>output</code> parameter is the directory of output.</p>
<div class="sourceCode" id="cb13"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/MicrobiotaProcess/man/ggdifftaxbar.html">ggdifftaxbar</a></span><span class="op">(</span>obj<span class="op">=</span><span class="va">deres</span>, xtextsize<span class="op">=</span><span class="fl">1.5</span>, 
             output<span class="op">=</span><span class="st">"IBD_biomarkder_barplot"</span>,
             coloslist<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#00AED7"</span>, <span class="st">"#FD9347"</span><span class="op">)</span><span class="op">)</span></pre></div>
</div>
<div id="example-of-visualization-of-different-results-by-ggeffectsize" class="section level2">
<h2 class="hasAnchor">
<a href="#example-of-visualization-of-different-results-by-ggeffectsize" class="anchor"></a>Example of visualization of different results by <code>ggeffectsize</code>
</h2>
<p>The result is similar with the result of <code>ggdiffbox</code>, the bigger confident interval shows that the biomarker is more fluctuant owing to the influence of samples number.</p>
<div class="sourceCode" id="cb14"><pre class="downlit">
<span class="va">es_p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MicrobiotaProcess/man/ggeffectsize.html">ggeffectsize</a></span><span class="op">(</span>obj<span class="op">=</span><span class="va">deres</span>, 
                     lineheight<span class="op">=</span><span class="fl">0.1</span>,
                     linewidth<span class="op">=</span><span class="fl">0.3</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#00AED7"</span>, 
                                    <span class="st">"#FD9347"</span><span class="op">)</span><span class="op">)</span> 

<span class="va">es_p</span></pre></div>
<div class="figure">
<img src="MicrobiotaProcessWorkshop_files/figure-html/plotES-1.png" alt="\label{fig:ESplot}Example of effect size of biomarker" width="480"><p class="caption">
Example of effect size of biomarker
</p>
</div>
</div>
<div id="session-information" class="section level2">
<h2 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h2>
<p>Here is the output of <code><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo()</a></code> on the system on which this document was compiled:</p>
<pre><code>#&gt; R version 4.0.2 (2020-06-22)
#&gt; Platform: x86_64-pc-linux-gnu (64-bit)
#&gt; Running under: Ubuntu 20.04 LTS
#&gt; 
#&gt; Matrix products: default
#&gt; BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so
#&gt; 
#&gt; locale:
#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
#&gt; 
#&gt; attached base packages:
#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     
#&gt; 
#&gt; other attached packages:
#&gt;  [1] gdtools_0.2.2            coin_1.3-1               survival_3.1-12         
#&gt;  [4] ggtree_2.3.6             patchwork_1.0.1          forcats_0.5.0           
#&gt;  [7] stringr_1.4.0            dplyr_1.0.2              purrr_0.3.4             
#&gt; [10] readr_1.4.0              tidyr_1.1.2              tibble_3.0.3            
#&gt; [13] tidyverse_1.3.0          ggplot2_3.3.2            phyloseq_1.33.0         
#&gt; [16] MicrobiotaProcess_1.1.10
#&gt; 
#&gt; loaded via a namespace (and not attached):
#&gt;   [1] TH.data_1.0-10      colorspace_1.4-1    ggsignif_0.6.0     
#&gt;   [4] ellipsis_0.3.1      modeltools_0.2-23   rprojroot_1.3-2    
#&gt;   [7] XVector_0.29.3      fs_1.5.0            aplot_0.0.6        
#&gt;  [10] rstudioapi_0.11     farver_2.0.3        ggrepel_0.8.2      
#&gt;  [13] bit64_4.0.5         fansi_0.4.1         lubridate_1.7.9    
#&gt;  [16] mvtnorm_1.1-1       xml2_1.3.2          codetools_0.2-16   
#&gt;  [19] splines_4.0.2       libcoin_1.0-6       knitr_1.30         
#&gt;  [22] ade4_1.7-15         jsonlite_1.7.1      broom_0.7.1        
#&gt;  [25] cluster_2.1.0       dbplyr_1.4.4        httr_1.4.2         
#&gt;  [28] BiocManager_1.30.10 rentrez_1.2.2       compiler_4.0.2     
#&gt;  [31] rvcheck_0.1.8       backports_1.1.10    assertthat_0.2.1   
#&gt;  [34] Matrix_1.2-18       lazyeval_0.2.2      cli_2.0.2          
#&gt;  [37] htmltools_0.5.0     tools_4.0.2         igraph_1.2.6       
#&gt;  [40] gtable_0.3.0        glue_1.4.2          reshape2_1.4.4     
#&gt;  [43] fastmatch_1.1-0     Rcpp_1.0.5          Biobase_2.49.1     
#&gt;  [46] cellranger_1.1.0    pkgdown_1.6.1       vctrs_0.3.4        
#&gt;  [49] Biostrings_2.57.2   rhdf5filters_1.1.3  multtest_2.45.0    
#&gt;  [52] svglite_1.2.3.2     ape_5.4-1           nlme_3.1-148       
#&gt;  [55] DECIPHER_2.17.1     iterators_1.0.12    xfun_0.18          
#&gt;  [58] Rmisc_1.5           rvest_0.3.6         lifecycle_0.2.0    
#&gt;  [61] phangorn_2.5.5      gtools_3.8.2        XML_3.99-0.5       
#&gt;  [64] zlibbioc_1.35.0     MASS_7.3-51.6       zoo_1.8-8          
#&gt;  [67] scales_1.1.1        hms_0.5.3           ragg_0.4.0         
#&gt;  [70] parallel_4.0.2      biomformat_1.17.0   sandwich_3.0-0     
#&gt;  [73] rhdf5_2.33.11       RColorBrewer_1.1-2  yaml_2.2.1         
#&gt;  [76] memoise_1.1.0       reshape_0.8.8       stringi_1.5.3      
#&gt;  [79] RSQLite_2.2.1       highr_0.8           S4Vectors_0.27.14  
#&gt;  [82] desc_1.2.0          foreach_1.5.0       randomForest_4.6-14
#&gt;  [85] tidytree_0.3.3      permute_0.9-5       BiocGenerics_0.35.4
#&gt;  [88] rlang_0.4.8         pkgconfig_2.0.3     systemfonts_0.3.2  
#&gt;  [91] matrixStats_0.57.0  evaluate_0.14       lattice_0.20-41    
#&gt;  [94] Rhdf5lib_1.11.3     labeling_0.3        treeio_1.13.1      
#&gt;  [97] bit_4.0.4           tidyselect_1.1.0    plyr_1.8.6         
#&gt; [100] magrittr_1.5        bookdown_0.20       R6_2.4.1           
#&gt; [103] IRanges_2.23.10     generics_0.0.2      multcomp_1.4-14    
#&gt; [106] DBI_1.1.0           haven_2.3.1         pillar_1.4.6       
#&gt; [109] withr_2.3.0         mgcv_1.8-31         modelr_0.1.8       
#&gt; [112] crayon_1.3.4        rmarkdown_2.4       readxl_1.3.1       
#&gt; [115] grid_4.0.2          data.table_1.13.0   blob_1.2.1         
#&gt; [118] vegan_2.5-6         reprex_0.3.0        digest_0.6.25      
#&gt; [121] textshaping_0.1.2   stats4_4.0.2        munsell_0.5.0      
#&gt; [124] quadprog_1.5-8</code></pre>
</div>
<div id="reference" class="section level2">
<h2 class="hasAnchor">
<a href="#reference" class="anchor"></a>Reference</h2>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Shuangbin Xu, Guangchuang Yu.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
